applications/ConvectionDiffusionApplication/custom_utilities/embedded_mls_constraint_process.cpp

```diff
@@ -0,0 +1,437 @@
+            auto& r_neigh_elems = rElement.GetValue(NEIGHBOUR_ELEMENTS);
+            for (std::size_t i_face = 0; i_face < n_faces; ++i_face) {
+                // The neighbour corresponding to the current face is ACTIVE means that the current face is surrogate boundary
+                // Flag the current neighbour owning the surrogate face as INTERFACE
+                auto& r_neigh_elem = r_neigh_elems[i_face];
+                if (r_neigh_elem.Is(ACTIVE)) {
+                    r_neigh_elem.Set(INTERFACE, true);
+                }
+            }
+        }
+    }
+}
+
+void EmbeddedMLSConstraintProcess::ReactivateElementsAndNodes()
+{
+    // Make intersected elements and their nodes ACTIVE
+    if ( !mDeactivateIntersectedElements ) {
+        for (auto& rElement : mpModelPart->Elements()) {
+            // Check if the element is split
+            const auto& r_geom = rElement.GetGeometry();
+            if (IsSplit(r_geom)) {
+                rElement.Set(ACTIVE, true);
+                for (auto& rNode : r_geom) {
+                    rNode.Set(ACTIVE, true);
+                }
+            }
+        }
+    }
+    // Make negative elements and their nodes ACTIVE
+    if ( !mDeactivateNegativeElements ) {
+        for (auto& rElement : mpModelPart->Elements()) {
+            // Check if the element is negative
+            const auto& r_geom = rElement.GetGeometry();
+            if (IsNegative(r_geom)) {
+                rElement.Set(ACTIVE, true);
+                for (auto& rNode : r_geom) {
+                    rNode.Set(ACTIVE, true);
+                }
+            }
+        }
+    }
+}
+
+void EmbeddedMLSConstraintProcess::ModifyDistances()
+{
+    auto& r_nodes = mpModelPart->Nodes();
+    double tol_d = 1.0e-12;
+
+    #pragma omp parallel for
+    for (int i = 0; i < static_cast<int>(r_nodes.size()); ++i) {
+        auto it_node = r_nodes.begin() + i;
+        double& d = it_node->FastGetSolutionStepValue(DISTANCE);
+
+        // Check if the distance values are close to zero, if so set the tolerance as distance value
+        if (std::abs(d) < tol_d) { 
+            d = (d > 0.0) ? tol_d : -tol_d; 
+        }
+    }
+}
+
+bool EmbeddedMLSConstraintProcess::IsSplit(const GeometryType& rGeometry)
+{
+    std::size_t n_neg = 0;
+    std::size_t n_pos = 0;
+    for (const auto& r_node : rGeometry) {
+        if (r_node.FastGetSolutionStepValue(DISTANCE) < 0.0) {
+            n_neg++;
+        } else {
+            n_pos++;
+        }
+    }
+    return (n_pos != 0 && n_neg != 0);
+}
+
+bool EmbeddedMLSConstraintProcess::IsNegative(const GeometryType& rGeometry)
+{
+    std::size_t n_neg = 0;
+    for (const auto& r_node : rGeometry) {
+        if (r_node.FastGetSolutionStepValue(DISTANCE) < 0.0) {
+            n_neg++;
+        }
+    }
+    return (n_neg == rGeometry.PointsNumber());
+}
+
+MLSShapeFunctionsFunctionType EmbeddedMLSConstraintProcess::GetMLSShapeFunctionsFunction()

```

Comment:
```suggestion
EmbeddedMLSConstraintProcess::MLSShapeFunctionsFunctionType EmbeddedMLSConstraintProcess::GetMLSShapeFunctionsFunction()

```
By doing this you can remove the `using` statement in the top of the file (to avoid the possibility of accessing it in the entire Kratos namespace).